### PR TITLE
test: test TOD against 3rd parties

### PR DESCRIPTION
**Pull request by phc1990**
✅ Formatting
✅ TOD <> GCRF testing:

- ✅ IAU 2000A vs STK
- ✅ IAU 2000A vs Orekit
- ✅ IAU 2000B vs STK
- ✅ IAU 2000B vs Orekit
- ✅ IAU 2006 vs STK
- ✅ IAU 2006 vs Orekit

### CODE REVIEW

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+#include <OpenSpaceToolkit/Mathematics/Geometry/3D/Transformations/Rotations/RotationVector.hpp>
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Frame.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Earth.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Derived/Angle.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Length.hpp>
+
+#include <Global.test.hpp>
+
+TEST(OpenSpaceToolkit_Physics_Coordinate_Frame_Providers_Transform, Test)
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:20]:
```suggestion
TEST(OpenSpaceToolkit_Physics_Coordinate_Frame_Providers_Transform, Validation)
```

Maybe?

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    using ostk::math::geom::d3::trf::rot::RotationMatrix;
+
+    using ostk::physics::units::Length;
+    using ostk::physics::units::Angle;
+    using ostk::physics::time::Scale;
+    using ostk::physics::time::Instant;
+    using ostk::physics::time::DateTime;
+    using ostk::physics::coord::Transform;
+    using ostk::physics::coord::Frame;
+    using ostk::physics::env::obj::celest::Earth;
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:45]: Move this at the top, outside of the test function. All test files should now comply with this ;)

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+             "TOD",
+             "IAU 2000A",
+             0.0,   // Position tolerance [m]
+             0.0,   // Velocity tolerance [m/s]
+             1e-4,  // Orientation axis tolerance []
+             1e-3,  // Orientation angle tolerance [rad]
+             0.0},  // Angular velocity tolerance [rad/s]
+
+            // GCRF <> TOD IAU 2000A (Orekit)
+
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:64]: Extra Line ToBeRemoved?

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+             "TOD",
+             "IAU 2000B",
+             0.0,   // Position tolerance [m]
+             0.0,   // Velocity tolerance [m/s]
+             1e-4,  // Orientation axis tolerance []
+             1e-3,  // Orientation angle tolerance [rad]
+             0.0},  // Angular velocity tolerance [rad/s]
+
+            // GCRF <> TOD IAU 2000B (Orekit)
+
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:89]: same

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+            const String otherStr = std::get<2>(referenceScenario);
+            const Real translationTolerance = std::get<3>(referenceScenario);
+            const Real velocityTolerance = std::get<4>(referenceScenario);
+            const Real orientationAxisTolerance = std::get<5>(referenceScenario);
+            const Real orientationAngleTolerance = std::get<6>(referenceScenario);
+            const Real angularVelocityTolerance = std::get<7>(referenceScenario);
+
+            const Table referenceData = Table::Load(referenceDataFile, Table::Format::CSV, true);
+
+            // Test
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:142]:
```suggestion
            // Comparison Test
```

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 6]
```diff
@@ -0,0 +1,331 @@
...
+                const Transform transformToFrame2 = frame1->getTransformTo(frame2, instant);
+
+                // Instant
+
+                {
+                    EXPECT_EQ(instant, transformToFrame1.getInstant());
+                    EXPECT_EQ(instant, transformToFrame2.getInstant());
+                }
+
+                // Translation
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:167]:
```suggestion
                // Relative Position
```

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+                        "{} @ {} Translation: {} [m] vs. {} [m] above {} [m] tolerance",
+                        scenario,
+                        instant.toString(Scale::TAI),
+                        referenceToFrame1.toString(),
+                        toFrame1.toString(),
+                        translationTolerance.toString()
+                    );
+                }
+
+                // Velocity
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:196]:
```suggestion
                // Relative Velocity
```

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 8]
```diff
@@ -0,0 +1,331 @@
...
+                        "{} @ {} Velocity: {} [m/s] vs. {} [m/s] above {} [m/s] tolerance",
+                        scenario,
+                        instant.toString(Scale::TAI),
+                        referenceToFrame1.toString(),
+                        toFrame1.toString(),
+                        velocityTolerance.toString()
+                    );
+                }
+
+                // Orientation
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:225]:
```suggestion
                // Relative Orientation
```

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 9]
```diff
@@ -0,0 +1,331 @@
...
+                               instant.toString(Scale::TAI),
+                               referenceToFrame1.getAxis().toString(),
+                               toFrame1.getAxis().toString(),
+                               orientationAxisTolerance.toString(),
+                               referenceToFrame1.getAngle().inRadians(),
+                               toFrame1.getAngle().inRadians()
+                           );
+                }
+
+                // Angular velocity
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:300]:
```suggestion
                // Relative Angular velocity
```

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+                        instant.toString(Scale::TAI),
+                        referenceToFrame1.toString(),
+                        toFrame1.toString(),
+                        angularVelocityTolerance.toString()
+                    );
+                }
+            }
+        }
+    }
+}
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:331]: extra line at bottom

File: src/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/IAU/Theory.cpp [hunk 11]
```diff
@@ -38,6 +38,24 @@ String StringFromTheory(const Theory& aTheory)
     return String::Empty();
 }
 
+Theory TheoryFromString(const String& aString)
```
Comment (Derollez) [inline src/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/IAU/Theory.cpp:41]: Add one unit test in cpp for this new function? :)

**Review by Derollez — COMMENTED** [review]
Awesome!

**Comment by Derollez** [comment]
Might be worth making the title a `test` here @phc1990

File: include/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/IAU/Theory.hpp [hunk 12]
```diff
@@ -41,6 +41,8 @@ enum class Theory
 
 String StringFromTheory(const Theory& aTheory);
```
[the remark this answers is not in the document]
Comment (phc1990) [reply include/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/IAU/Theory.hpp:43]: done for both

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp (hunk 1 above)
Comment (phc1990) [reply test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:20]: Yup

File: test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp (hunk 2 above)
Comment (phc1990) [reply test/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/Transform.test.cpp:45]: done!

**Pull request merged**